File: leetcode/dp/MaxiumSubarray.py

nums = [-2, 1, -3, 4, -1, 2, 1, -5, 4]
m = 0

# for i in range(len(nums)):
# sum = n]
# 	for j in range(i , len(nums)):
# 		sum += nums[j] 
# 		m = max(m,sum)

curr_sum = 0
start = 0 

for end in range(len(nums)):
	curr_sum += nums[end]
	
	m = max(curr_sum, m)

	if curr_sum < 0:
		curr_sum = 0
		start = end + 1


print(m)

File: leetcode/MaxiumSubarray.js
const nums = [-2, 1, -3, 4, -1, 2, 1, -5, 4];

//Brute Force Solution
let Max = 0;
for (let i = 0; i < nums.length; i++) {
  let sum = 0;
  for (let j = 0; j < nums.length; j++) {
    let elm = nums[j];
    sum += elm;
    Max = Math.max(sum, Max);
  }
}

console.log(Max);
